winpr/libwinpr/file/generic.c

```diff
@@ -828,6 +830,21 @@ int UnixChangeFileMode(const char* filename, int flags)
 	fl |= (flags & 0x0001) ? S_IXOTH : 0;
 	return chmod(filename, fl);
 #else
-	return 0;
+	int rc;
+	WCHAR *wfl = NULL;
+
+	int fl = 0;
+
+	ConvertToUnicode(CP_UTF8, 0, filename, -1, &wfl, 0);
+	if (!wfl)
+		return -1;
+
```

Comment:
Although this check works it was suggested to check the return value of ConvertToUnicode.